ctfcli/cli/challenges.py

```diff
@@ -178,22 +185,11 @@ def update(self, challenge=None):
             if challenge and challenge != folder:
                 continue
             if url.endswith(".git"):
-                click.echo(f"Cloning {url} to {folder}")
-                subprocess.call(["git", "init"], cwd=folder)
-                subprocess.call(["git", "remote", "add", "origin", url], cwd=folder)
-                subprocess.call(["git", "add", "-A"], cwd=folder)
-                subprocess.call(
-                    ["git", "commit", "-m", "Persist local changes (ctfcli)"],
-                    cwd=folder,
-                )
-                subprocess.call(
-                    ["git", "pull", "--allow-unrelated-histories", "origin", "master"],
-                    cwd=folder,
-                )
+                click.echo(f"Pulling latest {url} to {folder}")
+                subprocess.call(["git", "subtree", "pull", "--prefix", folder, url, "master", "--squash"])
                 subprocess.call(["git", "mergetool"], cwd=folder)
                 subprocess.call(["git", "clean", "-f"], cwd=folder)
                 subprocess.call(["git", "commit", "--no-edit"], cwd=folder)
```

Comment:
Does this flow still work? We might want to consider allowing the specifying of the tool. I don't like using vimdiff for this.

Reply:
The flow works in that if you resolve the merge conflict it does update. The merge tool doesn't open in terminal though, in my case due to an XCode error. 
By specifying the merge tool, are you thinking of a click.prompt()? I tried the --prompt option and that did not offer a a chance to select a tool.